jobs/elasticsearch-platform/templates/bin/pre-start

```diff
@@ -36,6 +36,15 @@ fi
 
 <% if p("elasticsearch.migrate_data") %>
 if [[ -d /var/vcap/store/elasticsearch ]]; then
-    cp -pRu /var/vcap/store/elasticsearch/nodes /var/vcap/store/elasticsearch-platform
+  mkdir /var/vcap/store/elasticsearch-platform || true
+  rsync -au --delete /var/vcap/store/elasticsearch/nodes /var/vcap/store/elasticsearch-platform/nodes
+fi
+<% end %>
+
+<% if p("elasticsearch.delete_migrated_data") %>
+if [[ -d /var/vcap/store/elasticsearch && -d /var/vcap/store/elasticsearch-platform ]]; then
+  if cat /var/vcap/jobs/elasticsearch-platform/config/elasticsearch.yml | grep -q 'data: "/var/vcap/store/elasticsearch-platform"'; then
+    rm -rf /var/vcap/store/elasticsearch
```

Comment:
I've gone back and forth on whether including this is a good idea. I decided to include it because:

- Manually running `rm -rf` on any VM scares me given the possibility of making a mistake
- Surrounding this command in extra test conditions in code adds a layer of safety
  - We're testing that both the original and new job name data directories exist AND that the config for the new job is pointing at the new data directory. So we shouldn't be able to accidentally delete the `elasticsearch` data while the Elasticsearch config is still using it. I've run the `if` conditions manually in the VM to confirm that they evaluate as expected.
- We can disable/enable it through BOSH config

Reply:
👍